### PR TITLE
feat: modify nova test trait

### PR DESCRIPTION
**Pull request by AZabolotnikov**
modify NovaTestTrait for updated requipments nova test generator

### CODE REVIEW

File: src/Tests/NovaTestTrait.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    protected function novaGetCreationFields(string $resourceClass): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->getJson("{$resourceUri}/creation-fields");
+    }
+
+    protected function novaRunAction(string $resourceClass, string $action, ?array $request = []): TestResponse
```
Comment (DenTray) [inline src/Tests/NovaTestTrait.php:56]:
```suggestion
    protected function novaRunAction(string $resourceClass, string $actionClass, ?array $request = []): TestResponse
```

File: src/Tests/NovaTestTrait.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    protected function novaSearchParams(array $filters, string $search = '', int $perPage = 25): array
+    {
+        return [
+            'search' => $search,
+            'filters' => base64_encode(json_encode($filters)),
+            'perPage' => $perPage,
+        ];
+    }
+
+    protected function novaCreateResource(string $resourceClass, ?array $data = []): TestResponse
```
Comment (DenTray) [inline src/Tests/NovaTestTrait.php:21]: let's add postfix to all api call helpers

```suggestion
    protected function novaCreateResourceAPICall(string $resourceClass, ?array $data = []): TestResponse
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Traits/NovaTestTrait.php [hunk 3]
```diff
@@ -2,14 +2,108 @@
...
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->getJson("{$resourceUri}/{$resourceId}", $data);
+    }
+
+    protected function novaSearchResourceAPICall(string $resourceClass, ?array $request = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        return $this->json('get', $resourceUri, $request);
```
Comment (DenTray) [inline src/Traits/NovaTestTrait.php:46]:
```suggestion
        return $this->getJson($resourceUri, $request);
```

File: src/Traits/NovaTestTrait.php [hunk 4]
```diff
@@ -2,14 +2,108 @@
...
+        return $this->getJson("{$resourceUri}/creation-fields");
+    }
+
+    protected function novaRunActionAPICall(string $resourceClass, string $actionClass, ?array $request = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        $actionUri = app($actionClass)->uriKey();
+
+        return $this->json('POST', "{$resourceUri}/action?action={$actionUri}", $request);
```
Comment (DenTray) [inline src/Traits/NovaTestTrait.php:62]:
```suggestion
        return $this->postJson("{$resourceUri}/action?action={$actionUri}", $request);
```

File: src/Traits/NovaTestTrait.php [hunk 5]
```diff
@@ -2,14 +2,108 @@
...
+
+    protected function novaGetActionsAPICall(string $resourceClass, array $resourceIds): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
+
+        $request = [
+            'resources' => implode(',', $resourceIds),
+        ];
+
+        return $this->json('get', "{$resourceUri}/actions", $request);
```
Comment (DenTray) [inline src/Traits/NovaTestTrait.php:73]:
```suggestion
        return $this->getJson("{$resourceUri}/actions", $request);
```

File: src/Traits/NovaTestTrait.php [hunk 6]
```diff
@@ -2,14 +2,108 @@
...
+    }
+
+    protected function novaActingAs(?Authenticatable $user = null): TestCase|self
+    {
+        return (empty($user))
+            ? $this
+            : $this->actingAs($user, 'web');
+    }
+
+    protected function getNovaResourceUri(string $modelClass): string
```
Comment (DenTray) [inline src/Traits/NovaTestTrait.php:101]:
```suggestion
    protected function generateNovaUri(string $modelClass, string $path = ''): string
```

File: src/Traits/NovaTestTrait.php [hunk 7]
```diff
@@ -2,14 +2,108 @@
...
         return [
             'search' => $search,
             'filters' => base64_encode(json_encode($filters)),
             'perPage' => $perPage,
         ];
     }
+
+    protected function novaCreateResourceAPICall(string $resourceClass, ?array $data = []): TestResponse
+    {
+        $resourceUri = $this->getNovaResourceUri($resourceClass);
```
Comment (DenTray) [inline src/Traits/NovaTestTrait.php:23]:
```suggestion
        $uri = $this->getNovaResourceUri($resourceClass);
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Traits/NovaTestTrait.php (hunk 4 above)
Comment (AZabolotnikov) [reply src/Traits/NovaTestTrait.php:62]: As strange as it may sound, tests won't work like that, I checked

**Review by DenTray — APPROVED** [review]

**Pull request merged**